env/env_custom.py

```diff
@@ -37,11 +37,17 @@ def __init__(self, df, main_tickers, all_tickers, max_assets_amount_per_trade, t
         self.comission_value = comission_value
         # Threshold from a Uniform distribution to add noise
         self.turbulence_threshold = turbulence_threshold
+        self.discrete_actionspace = discrete_actionspace
 
         self.state = State(technical_indicator_list)
         self.portfolio = Portfolio(cash=initial_amount, ticker_list=self.main_tickers)
-        self.action_space = spaces.Box(low=-1, high=1, shape=(len(self.main_tickers),))
-        self.observation_space = spaces.Box(low=-np.inf, high=np.inf,
+        if self.discrete_actionspace==False:
+            self.action_space = spaces.Box(low=-1, high=1, shape=(len(self.main_tickers),))
+            self.observation_space = spaces.Box(low=-np.inf, high=np.inf,
+                                            shape=(self.state.get_size(self.main_tickers, self.all_tickers),))
+        else:
+            self.action_space = spaces.Discrete(3)
+            self.observation_space = spaces.Box(low=-np.inf, high=np.inf,
```

Comment:
This seems to be the same in both cases. I would put it before the if condition. What do you think?